integrations/pgvector/src/haystack_integrations/document_stores/pgvector/document_store.py

```diff
@@ -307,7 +331,12 @@ def write_documents(self, documents: List[Document], policy: DuplicatePolicy = D
             raise DuplicateDocumentError from ie
         except Error as e:
             self._connection.rollback()
-            raise DocumentStoreError from e
+            sql_query_str = sql_insert.as_string(self._cursor)
+            error_msg = (
+                f"Could not write documents to PgvectorDocumentStore. \n"
+                f"SQL query: {sql_query_str} \nParameters: {db_documents}"
+            )
+            raise DocumentStoreError(error_msg) from e
```

Comment:
Same security considerations

Reply:
I've been thinking about it a bit. What do you suggest to show useful information without risk of leaks?